Pensel/firmware/modules/utilities/scheduler.c

```diff
@@ -0,0 +1,118 @@
+/*!
+ * @file    scheduler.c
+ * @author  Tyler Holmes
+ * @version 0.1.0
+ * @date    28-April-2018
+ * @brief   A rudamentary scheduler
+ */
+#include <stdint.h>
+#include <stdbool.h>
+
+#include "common.h"
+#include "scheduler.h"
+
+// TODO remove - debug
+// #include "peripherals/UART/UART.h"
+
+
+ret_t scheduler_init(schedule_t * schedule)
+{
+    schedule->num_slots_used = 0;
+    for (uint8_t i=0; i < NUM_SLOTS; i++) {
+        schedule->callbacks[i] = NULL;
+        schedule->callback_times_ms[i] = 0;
+        schedule->callback_last_run_ms[i] = 0;
+    }
+    return RET_OK;
+}
+
+
+ret_t scheduler_add(schedule_t * schedule,
+                    int32_t callback_time_ms,
+                    ret_t (*callback)(int32_t * callback_time_ms),
+                    uint8_t * schedule_id)
+{
+    bool slot_found = false;
```

Comment:
You know the capacity of your schedule array, and the number of filled slots, so is this variable really necessary when you could just stick with the check you have on line 36?

Reply:
Yeah I suppose the if check around that var will never be reached!